evennia/utils/ansi.py

```diff
@@ -733,6 +736,23 @@ def __new__(cls, *args, **kwargs):
     def __str__(self):
         return self._raw_string
 
+    def __format__(self, format_spec):
+        format_data = self.re_format.match(format_spec).groupdict()
```

Comment:
This definitely needs a doc string explaining the purpose of this new method. Including some examples.